### PR TITLE
Fixed ensemble documentation

### PR DESCRIPTION
**Pull request by erhart1**
This PR introduces a rearrangement of the documentation of the `ensemble` keyword (see below).
It also fixes the compilation of the documentation, which was broken due to a sphinx syntax error.

The page for the `ensemble` keyword had grown very long and difficult to navigate.
This became in particular apparent after the addition of the MTTK integrators in a recent PR.
This PR therefore splits the documentation of the `ensemble` keyword over four pages, describing respectively
* "standard" ensembles (SCR, Langevin, microcanonical etc)
* MTTK integrators
* methods for studying thermal transport
* path integral MD

### CODE REVIEW

File: doc/gpumd/input_parameters/ensemble_heat.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.. _kw_ensemble_heat:
+
+:attr:`ensemble` (thermal transport)
+====================================
+
+The :attr:`ensemble` keyword is used to set up an integration method (an integrator).
+The integrators described on this page provide methods for studying thermal transport via classical :term:`MD` simulations.
+Background information pertaining to the methods referred to on this page can be found :ref:`here <green_kubo_method>`.
```
Comment (brucefan1983) [inline doc/gpumd/input_parameters/ensemble_heat.rst:8]: The `heat` versions are used for NEMD methods not Green-Kubo.

File: doc/gpumd/input_parameters/ensemble_mttk.rst [hunk 2]
```diff
@@ -1,98 +1,104 @@
...
 
     ensemble npt_mttk temp 300 300 iso 10 10
 
-This command set the target temperature to 300 K and the target pressure to 10 GPa.
-The cell's shape will not change during the simlation, only the volume will change.
-It is suitable for simualting liquid.
-If not constrained, the cell shape may undergo extreme changes since liquids are not elastic.
+This command sets the target temperature to 300 K and the target pressure to 10 GPa.
+The cell shape will not change during the simlation but only the volume.
+These conditions are is suitable for simulating liquids.
```
Comment (brucefan1983) [inline doc/gpumd/input_parameters/ensemble_mttk.rst:60]: redundant "is"

**Comment by brucefan1983** [comment]
Thanks for the re-factoring, which is very nice!

**Comment by brucefan1983** [comment]
It's ready to merge after small changes.

File: doc/gpumd/input_parameters/ensemble_heat.rst (hunk 1 above)
Comment (erhart1) [reply doc/gpumd/input_parameters/ensemble_heat.rst:8]: Those methods are all on the same page and the reference is shared.

File: doc/gpumd/input_parameters/ensemble_mttk.rst (hunk 2 above)
Comment (erhart1) [reply doc/gpumd/input_parameters/ensemble_mttk.rst:60]:
```suggestion
These conditions are suitable for simulating liquids.
```

**Comment by erhart1** [comment]
The homepage is updated now
https://gpumd.org

**Pull request merged**